includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -537,6 +537,12 @@ public function has_allowed_items_in_cart() {
 			}
 		}
 
+		// We don't support multiple packages with Payment Request Buttons because we can't offer a good UX.
+		$packages = WC()->cart->get_shipping_packages();
+		if ( 1 < count( $packages ) ) {
+			return false;
+		}
+
```

Comment:
There's some more discussion in p1626166679419900-slack-CU6SYV31A on why we think this check is sufficient.